sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala

```diff
@@ -77,6 +77,19 @@ class IntervalUtilsSuite extends SparkFunSuite with SQLHelper {
     }
   }
 
+  test("string to interval: interval with dangling parts should not results null") {
+    checkFromInvalidString("+", "dangling interval value sign '+'")
+    checkFromInvalidString("-", "dangling interval value sign '-'")
+    checkFromInvalidString("+ 2", "dangling interval value '2'")
+    checkFromInvalidString("- 1", "dangling interval value '1'")
+    checkFromInvalidString("1", "dangling interval value '1'")
```

Comment:
how about `Expect a unit after '1' but hit EOL`

Reply:
done, but looks a bit weird for `interval '-.'` to expect a unit name